### PR TITLE
Detail header layout

### PR DESCRIPTION
**Pull request by acz1992**

### CODE REVIEW

**Comment by acz1992** [comment]
ready for review.

File: src/components/DetailHeaderLayOut.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <header
+      className="flex items-center justify-between
+     bg-dark"
+    >
+      <Link to={`${toAddress}`}>
+        <ChevronLeftIcon className="self-center h-6 fill-white" />
+      </Link>
+
+      <h1 className="text-title">{title}</h1>
+      <div></div>
```
Comment (NoHara42) [inline src/components/DetailHeaderLayOut.tsx:20]: empty div?

File: src/layouts/MoviesLayout.tsx [hunk 2]
```diff
@@ -3,8 +3,8 @@ import { Outlet } from "react-router-dom";
 export function MovieLayout() {
   return (
     <div>
-      <header>chevronBack</header>
       <Outlet />
+      <p>Movie layout</p>
```
Comment (NoHara42) [inline src/layouts/MoviesLayout.tsx:7]: this seems a bit literal?

File: src/layouts/RootLayout.tsx [hunk 3]
```diff
@@ -13,11 +13,10 @@ const routes = [
   {
     name: "Movies",
     path: "/movies",
-    children: [],
   },
   {
     name: "MovieDetails",
-    path: "/movies/details",
+    path: "/movieid/details",
```
Comment (NoHara42) [inline src/layouts/RootLayout.tsx:19]: wouldn't "/:movieid/details" fit the example in the documentation?

https://reactrouter.com/en/main/routers/create-browser-router

File: src/routes/CastCrew.tsx [hunk 4]
```diff
@@ -1,7 +1,9 @@
...
+
 export function CastCrew(): JSX.Element {
-  return (
-    <div>
-      <h1>Cast and Crew</h1>
-    </div>
-  );
+  const detailHeaderProps = {
+    title: "Cast and Crew",
+    toAddress: "../movie/details",
```
Comment (NoHara42) [inline src/routes/CastCrew.tsx:6]: don't you need the `movieId` to be interpolated into this string in place of movie?

Think about using `useParams()` to get this id
https://reactrouter.com/en/main/hooks/use-params

File: src/routes/MovieDetails.tsx [hunk 5]
```diff
@@ -11,10 +12,16 @@ export async function loadMovieDetails({ params }: props): Promise<string> {
 }
 
 export function MovieDetails(): JSX.Element {
-  const movie: string = useLoaderData();
+  /* const movie: string = useLoaderData(); */
+
+  const detailHeaderProps = {
+    title: "Movie Details",
+    toAddress: "../",
```
Comment (NoHara42) [inline src/routes/MovieDetails.tsx:19]: have you tested that this works?

File: src/routes/SelectTime.tsx [hunk 6]
```diff
@@ -1,4 +1,11 @@
+import { DetailHeader } from "../components/DetailHeaderLayOut";
+
 export function SelectTime(): JSX.Element {
+  const detailHeaderProps = {
+    title: "Select Time",
+    toAddress: "../..",
```
Comment (NoHara42) [inline src/routes/SelectTime.tsx:6]: does this go back to the homepage? shouldn't it go back to the movie details page?

**Review by NoHara42 — CHANGES_REQUESTED** [review]
just a few changes left @acz1992 
and remember to reverse merge when you're finished

File: src/layouts/MoviesLayout.tsx (hunk 2 above)
Comment (acz1992) [reply src/layouts/MoviesLayout.tsx:7]: This was just for sign posting for now, Maria is working on the Movies Layout page, so she will change this

**Comment by ThoJah** [comment]
Instead of reusing the DetailHeader() with fixed URL's passed in as a argument, we use it only once now in MovieLayout() with an onClick={ ( ) => navigate(-1) }
When trying navigate(-1) before we also implemented the component on every page, which led to endless cycles.
Having it rendered only once solved that problem

Ready for Review

File: src/components/DetailHeaderLayOut.tsx (hunk 1 above)
Comment (ThoJah) [reply src/components/DetailHeaderLayOut.tsx]: We use this div to distribute the space in the header equally, added a width now that matches the width of the backChevron

**Review by dnmct — APPROVED** [review]

**Pull request merged**